mne/viz/_3d.py

```diff
@@ -1227,19 +1233,46 @@ def _plot_sensors(renderer, info, to_cf_t, picks, meg, eeg, fnirs,
         if len(locs[sensor_type]) > 0:
             sens_loc = np.array(locs[sensor_type])
             sens_loc = sens_loc[~np.isnan(sens_loc).any(axis=1)]
-            color = defaults[sensor_type + '_color']
             scale = defaults[sensor_type + '_scale']
-            actor, _ = _plot_glyphs(renderer=renderer, loc=sens_loc * scalar,
-                                    color=color, scale=scale * scalar,
-                                    opacity=sensor_opacity,
-                                    orient_glyphs=orient_glyphs,
-                                    scale_by_distance=scale_by_distance,
-                                    project_points=project_points,
-                                    surf=surf, check_inside=check_inside,
-                                    nearest=nearest)
-            if sensor_type in ('source', 'detector'):
-                sensor_type = 'fnirs'
-            actors[sensor_type].append(actor)
+            if sensor_colors is None:
+                color = defaults[sensor_type + '_color']
+                actor, _ = _plot_glyphs(
+                    renderer=renderer,
+                    loc=sens_loc * scalar,
+                    color=color,
+                    scale=scale * scalar,
+                    opacity=sensor_opacity,
+                    orient_glyphs=orient_glyphs,
+                    scale_by_distance=scale_by_distance,
+                    project_points=project_points,
+                    surf=surf, check_inside=check_inside,
+                    nearest=nearest
+                )
+                if sensor_type in ('source', 'detector'):
+                    sensor_type = 'fnirs'
+                actors[sensor_type].append(actor)
+            else:
+                actor_list = []
+                for idx_sen in range(sens_loc.shape[0]):
+                    if isinstance(sensor_colors, list):
```

Comment:
this check is still fragile / too stringent. a tuple of color strings will fail this check, and will then error out on the `else` clause.  I suggest something like this:

```py
sensor_colors = np.asarray(sensor_colors)
if sensor_colors.ndim not in (1, 2) or sensor_colors.shape[0] != sens_loc.shape[0]:
    raise ValueError('insert error message about expected shape and actual shape we got')
color = sensor_colors[idx_sen]
# ...then the call to _plot_glyphs()
```